qualcommax: ipq60xx: add Qihoo 360V6 support

aiamadeus opened the pull request:
Specifications:
  SoC:     Qualcomm IPQ6000 1.5GHz
  RAM:     NTCC256M16ER-EK 512MiB
  Flash:   W29N01HZSINA 128MiB
  ETH:     QCA8075 (3x LAN, 1x WAN)
  WLAN1:   2.4GHz 802.11b/g/n/ax 2x2
  WLAN2:   5GHz 802.11a/n/ac/ax 2x2
  Power:   DC 12V 1.5A
  Button:  Reset, Wps
  USB:     1x 2.0

Flash instructions:
  1. Download the initramfs image, rename it to
     initramfs.itb, host it with the tftp server.
  2. Interrupt U-Boot and run these commands:
     tftpboot initramfs.itb
     bootm
  3. After openwrt boots up, use scp or luci web
     to upload sysupgrade.bin to upgrade.

aiamadeus committed:
qualcommax: ipq60xx: add Qihoo 360V6 support

Specifications:
  SoC:     Qualcomm IPQ6000 1.5GHz
  RAM:     NTCC256M16ER-EK 512MiB
  Flash:   W29N01HZSINA 128MiB
  ETH:     QCA8075 (3x LAN, 1x WAN)
  WLAN1:   2.4GHz 802.11b/g/n/ax 2x2
  WLAN2:   5GHz 802.11a/n/ac/ax 2x2
  Power:   DC 12V 1.5A
  Button:  Reset, Wps
  USB:     1x 2.0

Flash instructions:
  1. Download the initramfs image, rename it to
     initramfs.itb, host it with the tftp server.
  2. Interrupt U-Boot and run these commands:
     tftpboot initramfs.itb
     bootm
  3. After openwrt boots up, use scp or luci web
     to upload sysupgrade.bin to upgrade.

Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
Link: https://github.com/openwrt/openwrt/pull/15940
Signed-off-by: Robert Marko <robimarko@gmail.com>

aiamadeus committed:
qualcommax: image: cleanup whitespace

Replace whitespaces with tab.

Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
Link: https://github.com/openwrt/openwrt/pull/15940
Signed-off-by: Robert Marko <robimarko@gmail.com>

breeze303 commented:
Why did it take so long to review ipq60xx official like constipation？

aiamadeus commented:
@robimarko Could you take a look at this?

robimarko commented on target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:
```diff
@@ -29,9 +32,30 @@ ipq60xx_setup_interfaces()
+		lan_mac=$(mtd_get_mac_ascii factory lanMac)
+		wan_mac=$(macaddr_add "$lan_mac" 1)
+		label_mac=$lan_mac
+		;;
```
This should be replaced via NVMEM which now can read ASCII MAC-s as well

robimarko replied:
It doesnt have to be, you can use NVMEM on SMEM partition tables.

aiamadeus replied:
Do you mean something like this?
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";

			partition-0-factory {
				compatible = "fixed-partitions";
				label = "factory";
				read-only;
				#address-cells = <1>;
				#size-cells = <1>;

				partition@0 {
					reg = <0x0 0x40000>;

					macaddr_factory_ethaddr: lanMac {};
				};
			};
		};
	};
```

robimarko replied:
Close, but you cannot use fixed partitions on it:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/Documentation/devicetree/bindings/mtd/partitions/qcom,smem-part.yaml?h=next-20241128#n45

aiamadeus replied:
Unfortunately NVMEM on SMEM partition table does not work:
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";
			#address-cells = <1>;
			#size-cells = <1>;

			partition-factory {
				compatible = "nvmem-cells";
				#address-cells = <1>;
				#size-cells = <1>;
				label = "factory";
				read-only;

				macaddr_lan: macaddr@47 {
					compatible = "mac-base";
					reg = <0x47 0x11>;
					#nvmem-cell-cells = <1>;
				};
			};
		};
	};
```

robimarko replied:
Ahh, its u-boot-env variable, yeah in that case only ethaddr, eth1addr and so on are parseable for MAC-s

aiamadeus replied:
https://github.com/openwrt/openwrt/blob/39b0f081ecaf4fb9acc622d694dc52293a4dda1f/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch#L170-L174

It's actually just "ethaddr", so can we keep using the script?

robimarko replied:
Yes, untill U-Boot driver gets converted to layouts

aiamadeus committed:
ipq-wifi: update to version 2024-11-28

Contains following updates:

 * ipq8074: add TP-Link_deco-x80-5g BDF
 * ipq6018: add BDF for Qihoo 360V6

Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
Link: https://github.com/openwrt/openwrt/pull/15940
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Thanks! Rebased on top of main and merged!
